proposals/2024/07-17_container-outputs.md

```diff
@@ -0,0 +1,394 @@
+# Platform Settings
+
+<!--
+The name of this markdown file should:
+
+1. Short and contain no more then 30 characters
+
+2. Contain the date of submission in MM-DD format
+
+3. Clearly state what the proposal is being submitted for
+-->
+
+| Key           | Value |
+| :-----------: | :-: |
+| **Author(s)** | Easton Crupper |
+| **Reviewers** |  |
+| **Date**      | July 17th, 2024 |
+| **Status**    | In Progress |
+
+<!--
+If you're already working with someone, please add them to the proper author/reviewer category.
+
+If not, please leave the reviewer category empty and someone from the Vela team will assign it to themself.
+
+Here is a brief explanation of the different proposal statuses:
+
+1. Reviewed: The proposal is currently under review or has been reviewed.
+
+2. Accepted: The proposal has been accepted and is ready for implementation.
+
+3. In Progress: An accepted proposal is being implemented by actual work.
+
+NOTE: The design is subject to change during this phase.
+
+4. Cancelled: While or before implementation the proposal was cancelled.
+
+NOTE: This can happen for a multitude of reasons.
+
+5. Complete: This feature/change is implemented.
+-->
+
+## Background
+
+<!--
+This section is intended to describe the new feature, redesign or refactor.
+-->
+
+**Please provide a summary of the new feature, redesign or refactor:**
+
+<!--
+Provide your description here.
+-->
+
+Consider the following scenario:
+
+A user wishes to leverage authentication generated in a previous step in a plugin. How does that look in Vela today?
+
+```yaml
+steps:
+    - name: generate credentials
+      image: alpine
+      commands:
+        - ./fetch-credentials.sh
+        - echo $FETCHED_CREDENTIALS > /vela/creds.txt
+
+    - name: plugin to use auth
+      image: vela/my-plugin:latest
+      parameters:
+        cred_path: /vela/creds.txt
+```
+
+This is the power of the shared Docker volume. All steps can tap into the workspace and "communicate" data. 
+
+However, there are a few shortcomings with this. Some are obvious; some are not.
+
+### Problem 1: environment variable data sourcing
+
+Say the user in the above example does not own or contribute to the `vela/my-plugin` plugin. Further, the plugin does not support `cred_path: <path>`; instead, it expects a token provided as `$PLUGIN_TOKEN`. How can we accommodate this in Vela today?
+
+```yaml
+steps:
+    - name: generate credentials
+      image: alpine
+      commands:
+        - ./fetch-credentials.sh
+        - echo $FETCHED_CREDENTIALS > /vela/creds.txt
+
+    - name: plugin to use auth
+      image: vela/my-plugin:latest
+      entrypoint:
+        - /bin/sh
+        - -c
+        - export PLUGIN_TOKEN=$(cat /vela/creds.txt); /bin/my-plugin
+```
```

Comment:
to drive the point home, i wouldn't mind seeing an example of the `after` version with the outputs container in play. wherever it makes the most sense. either that, or leading with that with "this is how one might expect it" but "this is actually the mess you have to go through to make it work today" part.